File: 4 semestre/Paradigmas de Linguagem de Programação/10_outubro/aula 16-10-25/questao 1/Usuario.java
/**
Paradigmas de Linguagens de Programação
Data: 16-10-25
*/

public class Usuario {
    private String nome;
    private String endereco;
    private String telefone;
    private String email;
    private String tipo; 
    private int idUsuario;

    public Usuario(String nome, String endereco, String telefone, String email, String tipo, int idUsuario) {
        this.nome = nome;
        this.endereco = endereco;
        this.telefone = telefone;
        this.email = email;
        this.tipo = tipo;
        this.idUsuario = idUsuario;
    }

    public void cadastrarUsuario() {
        System.out.println("Usuário " + nome + " cadastrado com sucesso!");
    }

    public void atualizarCadastro(String novoEndereco, String novoTelefone) {
        this.endereco = novoEndereco;
        this.telefone = novoTelefone;
        System.out.println("Cadastro atualizado com sucesso!");
    }

    public void consultarUsuario() {
        System.out.println("Usuário: " + nome + " | Tipo: " + tipo);
    }
}

File: 4_semestre/Paradigmas-de-Linguagem-de-Programacao/paradigmas-de-programação/orientado-a-objetos/paradigmapookotlin/Usuario.kt
package paradigmapookotlin

data class Usuario(
    val id:     String,
    val nome:   String,
    val email:  String,
    val tipo:   TipoUsuario
) {
    companion object {
        fun criarUsuarioAdmin(): Usuario {
            return Usuario("ADMIN-001", "Administrador", "admin@biblioteca.com", TipoUsuario.ADMIN)
        }
    }
}